### DISCUSSION TITLE
Essential Exception

### DISCUSSION THREAD

**Discussion by rachaelbradley** [w3c/wcag3#55] [General]
It would be easier to write outcomes in plain language if we used a universal Essential Exception instead of writing the essential exception into each Outcome. In WCAG 2 we add the essential exception directly to the Success Criteria.  

- Is there a way to write the essential exception so that it can be used across outcomes when applicable? 
- Does it make sense to work as if an Essential Exception exists and then revisit the question when we have more outcomes?

No voting is needed here but please add comments and thoughts. We will be discussing this at the 12 March 2024 meeting.

**Comment by mraccess77** [comment]
Essential may need to be scoped to a situations and there may also be situations where there should be no essential exception.  Otherwise people might apply it too broadly.  So it may be needed to put it at the outcome level.

**Reply by u9000-Nine** [reply]
I agree. There are some criteria which prevent physical harm to users, such as [2.3 Seizures and Physical Reactions](https://www.w3.org/TR/2023/REC-WCAG22-20231005/#seizures-and-physical-reactions), which should never fall under an allowed exception.

**Comment by alastc** [comment]
(Chair hat off) I would very much like a standard essential exception that applies to everything, it will make writing the outcomes & methods easier.

The current ['essential' definition in WCAG 2](https://www.w3.org/TR/WCAG22/#dfn-essential) is a little odd, combining two things together. 

I'd like to start with a simpler definition and see what the issues with that might be. For example:

> Where the information conveyed or the functionality provided cannot be achieved in another way that would conform, the '_outcome_' does not apply.

Also, I think it should stick to the core information / functionality, and not include legal aspects by default. I think the legal aspects should be dealt with on an outcome by outcome basis.

The scenarios where I've come across the essential exception being applied have been things like:

- Educational material where the point is to identify, visually, a particular thing (e.g. body part for medical purposes). Testing that someone can identify that body part visually is the point of the exercise, so alternative text is not helpful.
- Scientific graphs or maps, where a graduated scale over a map is necessary to show the data accurately.
- Memory games or tests.
- Games where reaction time is a key factor in the gameplay.
- Maps or visual representations which cannot be manipulated to fit a particular viewport, and/or the colour cannot be changed because they are supposed to represent the real life colours (e.g. national flags)
- Security considerations where re-typing in a password is necessary to avoid people getting locked out of an account. 

I'm sure there are others, those are ones I remember from WCAG 2.1 / 2.2 discussions.

**Reply by u9000-Nine** [reply]
> - Educational material where the point is to identify, visually, a particular thing (e.g. body part for medical purposes). Testing that someone can identify that body part visually is the point of the exercise, so alternative text is not helpful.

Unless the test is specifically an eye exam (or similar), I still think alt-text would be useful.

In your example, that seems more like a test of whether someone can identify body parts *based on it's appearance*, not specifically whether they can parse that appearance with their eyes.\* So in that situation, I think it should have alt-text that describes what the body part looks like.

\*I'm not exactly sure how to say that. I'm trying give leeway to people who might have poor vision or people with cognitive disabilities. In real life, people with poor vision might be able to identify the body part with a magnifying glass, so it might be that they simply need a higher-resolution image on the computer screen. I also know of some people with good vision who never the less rely on screen readers to prevent overwhelm (because of cognitive disabilities).

>  I would very much like a standard essential exception that applies to everything, it will make writing the outcomes & methods easier.

But there are also scenarios which should not have an essential exception, such as [2.3&nbsp;Seizures and Physical Reactions](https://www.w3.org/TR/2023/REC-WCAG22-20231005/#seizures-and-physical-reactions) and [1.4.2&nbsp;Audio Control](https://www.w3.org/TR/2023/REC-WCAG22-20231005/#audio-control)

**Reply by alastc** [reply]
> Unless the test is specifically an eye exam (or similar), I still think alt-text would be useful.

Sometimes, but if it is a test for Doctors on identifying a particular organ with 3 choices, asking "which one is the spleen?", then alt-text would probably be a give-away. I'm not sure how you could describe it without giving the answer away. 

> But there are also scenarios which should not have an essential exception, such as [2.3 Seizures and Physical Reactions](https://www.w3.org/TR/2023/REC-WCAG22-20231005/#seizures-and-physical-reactions) and [1.4.2 Audio Control](https://www.w3.org/TR/2023/REC-WCAG22-20231005/#audio-control)

Sure, but wouldn't it be easier to highlight those in the outcomes, and perhaps have a category similar to the '[non-interferance](https://www.w3.org/TR/WCAG22/#cc5)' in the conformance?

**Reply by u9000-Nine** [reply]
> Sometimes, but if it is a test for Doctors on identifying a particular organ with 3 choices, asking "which one is the spleen?", then alt-text would probably be a give-away. I'm not sure how you could describe it without giving the answer away.

You would just describe the way it looks using non-technical terms. It would then be as much of a giveaway as the the images. For example, if one of the images was a heart, you might describe it as "A red, squishy sack with tubes coming out of it". (I'm not a doctor and I'm describing a heart from memory, so take those specific words with a grain of salt. The actual alt-text could still be something similar, though, without giving away the answer.)

> Sure, but wouldn't it be easier to highlight those in the outcomes, and perhaps have a category similar to the '[non-interferance](https://www.w3.org/TR/WCAG22/#cc5)' in the conformance?

I suppose that could work, as long as the universal essential exception has wording to the effect of 'unless otherwise noted in the outcome (or whatever we choose)'.

That would also allow for more-specific handling of those outcomes. For example, we could write guidance on 2.3&nbsp;Seizures and Physical Reactions for how to handle training material for identifying such content (which will necessarily have examples of it)

**Comment by rachaelbradley** [comment]
### Summary 10 June 2025

Discussion in today's group touched on essential exceptions. How do we document them when the purpose of a product (example: Children's listening exercise) requires a specific ability? 

Points made:
- We do need to document a way to incorporate essential exceptions
- In the case of the example given, it may be that the product fails but policy/regulation acknowledges that the failure is acceptable (such as when an alternative, comparable experience is available).

**Reply by colabottles** [reply]
In my opinion, nothing regarding Safety or Deception (deceptive design and deceptive patterns) should ever have an "essential exception" due to the nature of these categories. Requirements will include statements on when and how these should apply in scope.

I think there are cases for essential exceptions but not in Safety or Deception. Will we have a document on which of the essential exceptions these apply to or will they be written into each guideline?

**Comment by CityMouse** [comment]
+1
